### PR TITLE
{CI} Add GitHub Action to Conditionally Trigger CI Based on Safe File Changes

### PR DESCRIPTION
**Pull request by wangzelin007**
This PR introduces a GitHub Action that conditionally triggers the CI process based on specific file changes in the PR. The purpose is to avoid automatically running CI when potentially risky files, such as configuration files (*.yml) or files within the scripts/ directory, are modified. This helps mitigate unintended impacts from sensitive file changes by not triggering CI in those cases.

image

**Related command**

**Description**

**Testing Guide**

**History Notes**

[Component Name 1] BREAKING CHANGE: `az command a`: Make some customer-facing breaking change
[Component Name 2] `az command b`: Add some customer-facing feature

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [ ] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
Thank you for your contribution! We will review the pull request and get back to you soon.

**Review by evelyn-ys — APPROVED** [review]

File: .github/workflows/CIGuardedTrigger.yml [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+
+        FILES=$(git --no-pager diff --name-only --diff-filter=ACMRT "$base_branch_pre"/"$base_branch"..."$diff_branch")
+        echo "Changed files:"
+        echo "$FILES"
+        
+        SHOULD_RUN_CI=true
+
+        # Check if any of the modified files are .yml or in scripts/ directory
+        for FILE in $FILES; do
+          if [[ "$FILE" == *.yml || "$FILE" == scripts/* ]]; then
```
Comment (bebound) [inline .github/workflows/CIGuardedTrigger.yml:42]: Any reason to skip `scripts/*`?

Comment (wangzelin007) [reply .github/workflows/CIGuardedTrigger.yml:42]: I think that non-CLI team members should not modify scripts. CLI team members modifying scripts and submitting PRs already automatically trigger the CI. If my understanding is incorrect, please correct me.

File: .github/workflows/CIGuardedTrigger.yml [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+      with:
+        fetch-depth: 0
+    
+    - name: Check changed files and conditionally trigger CI
+      env:
+        base_repo: ${{ github.event.pull_request.base.repo.clone_url }}
+        base_branch: ${{ github.event.pull_request.base.ref }}
+        base_branch_pre: "upstream"
+        diff_branch: ${{ github.event.pull_request.head.ref }}
+        REPO_TOKEN: ${{ secrets.CLI_BOT }}
```
Comment (zhoxing-ms) [inline .github/workflows/CIGuardedTrigger.yml:27]: Just out of curiosity, is the risk of exposing this token controllable?

Comment (wangzelin007) [reply .github/workflows/CIGuardedTrigger.yml:27]: This is a scoped token; it only has comment permissions for azure-cli and azure-cli-extensions.

Comment (zhoxing-ms) [reply .github/workflows/CIGuardedTrigger.yml:27]: Got it, thanks~

**Review by zhoxing-ms — APPROVED** [review]

**Review by bebound — APPROVED** [review]

**Review by MoChilia — CHANGES_REQUESTED** [review]
request change

**Review by Pan-Qi — APPROVED** [review]

**Pull request closed without merging**